website/blog/2024-10-25-llama-index-workflow/index.md

```diff
@@ -0,0 +1,423 @@
+            # Specify retrievers to use.
+            model_config={"retrievers": retrievers},
+            # Define dependency files to save along with the model
+            code_paths=["workflow"],
+            # Subdirectory to save artifacts (not important)
+            artifact_path="model",
+        )
+        models.append(model_info)
+```
+
+Now open the MLflow UI again, and this time it should show 4 MLflow Runs are recorded with different `retrievers` parameter values. By clicking each Run name and navigate to the "Artifacts" tab, you can see MLflow records the model and various metadata, such as dependency versions and settings.
+
+![MLflow Runs](llama_index_workflow_runs.png)
+
+
+## 7. Enable MLflow Tracing
+
+Before running the evaluation, there’s one final step: enabling **MLflow Tracing**. We'll dive into this feature and why we do this here later, but for now, you can enable it with a simple one-line command. MLflow will automatically trace every LlamaIndex execution.
+
+```python
+mlflow.llama_index.autolog()
+```
+
+## 8. Evaluate the Workflow with Different Retriever Strategies
+
+The example repository includes a sample evaluation dataset, `mlflow_qa_dataset.csv`, containing 30 question-answer pairs related to MLflow.
+
+```python
+import pandas as pd
+
+eval_df = pd.read_csv("data/mlflow_qa_dataset.csv")
+display(eval_df.head(3))
+```
+
+To evaluate the workflow, use the [mlflow.evaluate()](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.evaluate) API, which requires (1) your dataset, (2) the logged model, and (3) the metrics you want to compute.
+
+
+```python
+from mlflow.metrics import latency
+from mlflow.metrics.genai import answer_correctness, faithfulness
+
+
+for model_info in models:
+
+    with mlflow.start_run(run_id=model_info.run_id):
+        result = mlflow.evaluate(
+            # Pass the URI of the logged model above
+            model=model_info.model_uri,
+            data=eval_df,
+            # Specify the column for ground truth answers.
+            targets="ground_truth",
+            # Define the metrics to compute.
+            extra_metrics=[
+                latency(),
+                answer_correctness("openai:/gpt-4o-mini"),
+            ],
+            # The answer_correctness metric requires "inputs" column to be
+            # present in the dataset. We have "query" instead so need to
+            # specify the mapping in `evaluator_config` parameter.
+            evaluator_config={"col_mapping": {"inputs": "query"}},
+        )
+```
+
+In this example, we evaluate the model with two metrics:
+
+1. **Latency**: Measures the time taken to execute a workflow for a single query.
+2. **Answer Correctness**: Evaluates the accuracy of answers based on the ground truth, scored by the OpenAI GPT-4o model on a 1–5 scale.
+
+These metrics are just for demonstration purposes—you can add additional metrics like toxicity or faithfulness, or even create your own.
```

Comment:
Can we link to the metric creation guide? :)

Reply:
SG!